src/mcp/tools/dataconnect/execute_graphql.ts

```diff
@@ -33,13 +37,40 @@ export const execute_graphql = tool(
       requiresAuth: true,
     },
   },
-  async ({ query, serviceId, variables: unparsedVariables }, { projectId, config }) => {
+  async (
+    { query, serviceId, variables: unparsedVariables, useEmulator },
+    { projectId, config, emulatorHubClient },
+  ) => {
+    if (!projectId) {
```

Comment:
This is already checked here.

https://github.com/firebase/firebase-tools/blob/c64476f2a372d1d61c0fde10a0cf623ddd59b99f/src/mcp/index.ts#L147